src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -945,11 +956,12 @@ static int prepare_new_lease(
                 uint8_t hlen,
                 const uint8_t *chaddr,
                 be32_t gateway,
-                usec_t expiration) {
+                usec_t expiration,
+                char *hostname) {
 
         _cleanup_(dhcp_lease_freep) DHCPLease *lease = NULL;
 
-        lease = new(DHCPLease, 1);
+        lease = new0(DHCPLease, 1);
```

Comment:
Please drop the change. It is not necessary, as the object is initialized later by the line `*lease = (DHCPLease) {...};`.

Reply:
Reverted.

Ah right. I jad some segfault issues during testing due to ussing `strdupa` instead of `strdup` and this was an attempt of fixing it:)